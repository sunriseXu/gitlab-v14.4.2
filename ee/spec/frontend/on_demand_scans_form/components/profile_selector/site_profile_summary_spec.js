import { shallowMount } from '@vue/test-utils';
import App from 'ee/on_demand_scans_form/components/profile_selector/site_profile_summary.vue';
import { siteProfiles } from 'ee_jest/security_configuration/dast_profiles/mocks/mock_data';

describe('DastSiteProfileSummary', () => {
  let wrapper;

  const createWrapper = (profile = {}) => {
    wrapper = shallowMount(App, {
      propsData: {
        profile,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('renders properly', () => {
    it.each(siteProfiles)('profile $id $profileName', (profile) => {
      createWrapper(profile);

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
