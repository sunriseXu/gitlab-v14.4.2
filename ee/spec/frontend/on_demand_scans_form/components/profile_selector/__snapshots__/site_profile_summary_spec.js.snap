// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DastSiteProfileSummary renders properly profile gid://gitlab/DastSiteProfile/1 Non-validated 1`] = `
<div
  class="row"
>
  <summary-cell-stub
    class=""
    label="Target URL"
    value="http://none.test"
  />
   
  <summary-cell-stub
    label="Site type"
    value="Website"
  />
   
  <summary-cell-stub
    label="Authentication URL"
    value="http://none.test/sign-in"
  />
   
  <summary-cell-stub
    label="Username"
    value="non-validated@example.com"
  />
   
  <summary-cell-stub
    label="Password"
    value="••••••••"
  />
   
  <summary-cell-stub
    label="Username form field"
    value="session[username]"
  />
   
  <summary-cell-stub
    label="Password form field"
    value="session[password]"
  />
   
  <summary-cell-stub
    label="Submit button"
  />
   
  <summary-cell-stub
    label="Excluded URLs"
    value="http://none.test/sign-out,http://none.test/hidden"
  />
   
  <summary-cell-stub
    label="Request headers"
  />
   
  <summary-cell-stub
    label="Validation status"
    value="Not Validated"
  />
</div>
`;

exports[`DastSiteProfileSummary renders properly profile gid://gitlab/DastSiteProfile/2 Validation failed 1`] = `
<div
  class="row"
>
  <summary-cell-stub
    class=""
    label="Target URL"
    value="http://failed.test"
  />
   
  <summary-cell-stub
    label="Site type"
    value="Website"
  />
   
  <summary-cell-stub
    label="Authentication URL"
    value="http://failed.test/sign-in"
  />
   
  <summary-cell-stub
    label="Username"
    value="validation-failed@example.com"
  />
   
  <summary-cell-stub
    label="Password"
    value="••••••••"
  />
   
  <summary-cell-stub
    label="Username form field"
    value="session[username]"
  />
   
  <summary-cell-stub
    label="Password form field"
    value="session[password]"
  />
   
  <summary-cell-stub
    label="Submit button"
  />
   
  <summary-cell-stub
    label="Excluded URLs"
    value="http://failed.test/sign-out,http://failed.test/hidden"
  />
   
  <summary-cell-stub
    label="Request headers"
  />
   
  <summary-cell-stub
    label="Validation status"
    value="Not Validated"
  />
</div>
`;

exports[`DastSiteProfileSummary renders properly profile gid://gitlab/DastSiteProfile/3 Validation passed 1`] = `
<div
  class="row"
>
  <summary-cell-stub
    class=""
    label="Target URL"
    value="http://passed.test"
  />
   
  <summary-cell-stub
    label="Site type"
    value="Website"
  />
   
  <summary-cell-stub
    label="Authentication URL"
    value="http://passed.test/sign-in"
  />
   
  <summary-cell-stub
    label="Username"
    value="validation-passed@example.com"
  />
   
  <summary-cell-stub
    label="Password"
    value="••••••••"
  />
   
  <summary-cell-stub
    label="Username form field"
    value="session[username]"
  />
   
  <summary-cell-stub
    label="Password form field"
    value="session[password]"
  />
   
  <summary-cell-stub
    label="Submit button"
  />
   
  <summary-cell-stub
    label="Excluded URLs"
    value="http://passed.test/sign-out,http://passed.test/hidden"
  />
   
  <summary-cell-stub
    label="Request headers"
  />
   
  <summary-cell-stub
    label="Validation status"
    value="Validated"
  />
</div>
`;

exports[`DastSiteProfileSummary renders properly profile gid://gitlab/DastSiteProfile/4 Validation in progress 1`] = `
<div
  class="row"
>
  <summary-cell-stub
    class=""
    label="Target URL"
    value="http://inprogress.test"
  />
   
  <summary-cell-stub
    label="Site type"
    value="Website"
  />
   
  <summary-cell-stub
    label="Authentication URL"
    value="http://inprogress.test/sign-in"
  />
   
  <summary-cell-stub
    label="Username"
    value="validation-in-progress@example.com"
  />
   
  <summary-cell-stub
    label="Password"
    value="••••••••"
  />
   
  <summary-cell-stub
    label="Username form field"
    value="session[username]"
  />
   
  <summary-cell-stub
    label="Password form field"
    value="session[password]"
  />
   
  <summary-cell-stub
    label="Submit button"
  />
   
  <summary-cell-stub
    label="Excluded URLs"
    value="http://inprogress.test/sign-out,http://inprogress.test/hidden"
  />
   
  <summary-cell-stub
    label="Request headers"
  />
   
  <summary-cell-stub
    label="Validation status"
    value="Not Validated"
  />
</div>
`;

exports[`DastSiteProfileSummary renders properly profile gid://gitlab/DastSiteProfile/5 Validation pending 1`] = `
<div
  class="row"
>
  <summary-cell-stub
    class=""
    label="Target URL"
    value="http://pending.test"
  />
   
  <summary-cell-stub
    label="Site type"
    value="Website"
  />
   
  <summary-cell-stub
    label="Authentication URL"
    value="http://pending.test/sign-in"
  />
   
  <summary-cell-stub
    label="Username"
    value="validation-pending@example.com"
  />
   
  <summary-cell-stub
    label="Password"
    value="••••••••"
  />
   
  <summary-cell-stub
    label="Username form field"
    value="session[username]"
  />
   
  <summary-cell-stub
    label="Password form field"
    value="session[password]"
  />
   
  <summary-cell-stub
    label="Submit button"
  />
   
  <summary-cell-stub
    label="Excluded URLs"
    value="http://pending.test/sign-out,http://pending.test/hidden"
  />
   
  <summary-cell-stub
    label="Request headers"
  />
   
  <summary-cell-stub
    label="Validation status"
    value="Not Validated"
  />
</div>
`;
